components/brave_wallet/browser/brave_wallet_service.cc

```diff
@@ -715,7 +716,8 @@ void BraveWalletService::AddSuggestTokenRequest(
                    request->token->is_erc721, chain_id);
 
   if (!token && chain_id == mojom::kMainnetChainId)
-    token = BlockchainRegistry::GetInstance()->GetTokenByContract(addr);
+    token = BlockchainRegistry::GetInstance()->GetTokenByContract(
+        mojom::kMainnetChainId, addr);
```

Comment:
Probably not within this PR's scope, now that we'll have registry data for other chains, we should query the registry using chain_id for other chains too and use it if found.

Reply:
noted, thx

Reply:
Fixed in rebase here https://github.com/brave/brave-core/pull/11791/commits/60036d25b2dac7b15ee980dabb1b1bd366d447fb